searx/engines/core.py

```diff
@@ -23,92 +23,104 @@
 
 api_key = 'unset'
 
-base_url = 'https://core.ac.uk:443/api-v2/search/'
-search_string = '{query}?page={page}&pageSize={nb_per_page}&apiKey={apikey}'
-
+base_url = 'https://api.core.ac.uk/v3/search/works/'
 
 def request(query, params):
-
     if api_key == 'unset':
         raise SearxEngineAPIException('missing CORE API key')
 
-    search_path = search_string.format(
-        query=urlencode({'q': query}),
-        nb_per_page=nb_per_page,
-        page=params['pageno'],
-        apikey=api_key,
-    )
-    params['url'] = base_url + search_path
+    # API v3 uses different parameters
+    search_params = {
+        'q': query,
+        'offset': (params['pageno'] - 1) * nb_per_page,
+        'limit': nb_per_page,
+        'sort': 'relevance'
+    }
 
-    return params
+    params['url'] = base_url + '?' + urlencode(search_params)
+    params['headers'] = {'Authorization': f'Bearer {api_key}'}
 
+    return params
 
 def response(resp):
     results = []
     json_data = resp.json()
 
-    for result in json_data['data']:
-        source = result['_source']
+    for result in json_data.get('results', []):
+        # Get title
+        title = result.get('title')
+        if not title:
+            continue
+
+        # Get URL - try different options
         url = None
-        if source.get('urls'):
-            url = source['urls'][0].replace('http://', 'https://', 1)
+        
+        # Try DOI first
+        doi = result.get('doi')
+        if doi:
+            url = f'https://doi.org/{doi}'
 
-        if url is None and source.get('doi'):
+        if url is None and result.get('doi'):
             # use the DOI reference
-            url = 'https://doi.org/' + source['doi']
-
-        if url is None and source.get('downloadUrl'):
-            # use the downloadUrl
-            url = source['downloadUrl']
-
-        if url is None and source.get('identifiers'):
-            # try to find an ark id, see
-            # https://www.wikidata.org/wiki/Property:P8091
-            # and https://en.wikipedia.org/wiki/Archival_Resource_Key
-            arkids = [
-                identifier[5:]  # 5 is the length of "ark:/"
-                for identifier in source.get('identifiers')
-                if isinstance(identifier, str) and identifier.startswith('ark:/')
-            ]
-            if len(arkids) > 0:
-                url = 'https://n2t.net/' + arkids[0]
-
-        if url is None:
+            url = 'https://doi.org/' + str(result['doi'])
+
+        if url is None and result.get('id'):
+            url = 'https://core.ac.uk/works/' + str(result['id'])
+
+        # Then try downloadUrl
+        if not url and result.get('downloadUrl'):
+            url = result['downloadUrl']
+        
+        if not url and result.get('sourceFulltextUrls'):
+            url = result['sourceFulltextUrls']
+
+        # Skip if no URL found
+        if not url:
```

Comment:
```suggestion
            url = 'https://doi.org/' + str(result['doi'])
        elif result.get('id'):
            url = 'https://core.ac.uk/works/' + str(result['id'])
        elif result.get('downloadUrl'):
            url = result['downloadUrl']
        elif result.get('sourceFulltextUrls'):
            url = result['sourceFulltextUrls']
        else:
            continue
```
These should all be using `elif` statements, you can only ever get into one of them.